src/frontend/src/lib/components/carousel/Carousel.svelte

```diff
@@ -75,6 +81,12 @@
 
 		containerWidth = container.offsetWidth;
 
+		// Clear timers and stop further initialisation in case container is rendered but not visible (e.g. display: none)
+		if (containerWidth === 0) {
```

Comment:
Does that work in every browser and devices (including mobile phone)?

Should we plug quickly an intersection oberserve to be a on the safest side?

Reply:
Safari, Firefox, Chrome - ✅ 

I think IntersectionObserver could lead to some unwanted clearing/initialising processes - if Carousel is rendered and does not have display:none but still out of the screen (e.g. long tokens list on tablet and the carousel stays on top), `isIntersecting` will be giving false/true on scroll, so the events, timers, and etc. will re-initialise which is not really necessary. WDYT?

Reply:
Too some extention that can be considered as a feature. 😉
But I hear you.